app/lib/text_formatter.rb

```diff
@@ -93,48 +98,54 @@ def rewrite
   end
 
   def link_to_url(entity)
-    TextFormatter.shortened_link(entity[:url], rel_me: with_rel_me?)
+    MastodonOTELTracer.in_span('TextFormatter#link_to_url') do
+      TextFormatter.shortened_link(entity[:url], rel_me: with_rel_me?)
+    end
   end
 
   def link_to_hashtag(entity)
-    hashtag = entity[:hashtag]
-    url     = tag_url(hashtag)
+    MastodonOTELTracer.in_span('TextFormatter#link_to_hashtag') do
+      hashtag = entity[:hashtag]
+      url     = tag_url(hashtag)
 
-    <<~HTML.squish
-      <a href="#{h(url)}" class="mention hashtag" rel="tag">#<span>#{h(hashtag)}</span></a>
-    HTML
+      <<~HTML.squish
+        <a href="#{h(url)}" class="mention hashtag" rel="tag">#<span>#{h(hashtag)}</span></a>
+      HTML
+    end
   end
 
   def link_to_mention(entity)
-    username, domain = entity[:screen_name].split('@')
-    domain           = nil if local_domain?(domain)
-    account          = nil
-
-    if preloaded_accounts?
-      same_username_hits = 0
-
-      preloaded_accounts.each do |other_account|
-        same_username = other_account.username.casecmp(username).zero?
-        same_domain   = other_account.domain.nil? ? domain.nil? : other_account.domain.casecmp(domain)&.zero?
-
-        if same_username && !same_domain
-          same_username_hits += 1
-        elsif same_username && same_domain
-          account = other_account
+    MastodonOTELTracer.in_span('TextFormatter#link_to_mention') do
```

Comment:
Should we have an attribute that says if this used a preloaded account or not, to be able to know what %age are doing so?
And another one to say if this is a local account?

Reply:
We should have this information from the outermost span.

Reply:
We might want to add it here at some point, as far as I know there is no way to get cross-span data in Datadog, so for example you cannot compare performance of `link_to_mention` on "a parent span says the account is preloaded" vs the opposite.